xandikos/icalendar.py

```diff
@@ -505,9 +505,15 @@ def __repr__(self):
             self.negate_condition,
         )
 
+    def _ical_type(self, k):
+        if isinstance(k, self.type_fn):
+            return k
+        else:
+            return self.type_fn.from_ical(k)
+
     def match_indexes(self, indexes: SubIndexes):
         return any(
-            self.match(self.type_fn.from_ical(k)) for k in indexes[None])
```

Comment:
The indexes should never hold type_fn objects.

This can be simplified, it can just be "self.type_fn(self.type_fn.from_ical(k))"

Reply:
Then there is a bug somewhere else.  I checked both with category and summary, and it was `vText` and `vCategories`.

What's worse, when doing a `vCategories.from_ical(text_string)`, I got back ... a list, not a `vCategories` object :-(

This patch solves the issue for me at least.  If doing it in another way, I think the first priority would be to catch the problem in a unit test.